anomaly/anomaly_detector.py

```
@@ -255,12 +257,15 @@ def detect_anomaly(data: Dict) -> Tuple[bool, float]:
         if not _MODEL_LOADED:
             load_model()
 
-        # Validate input
-        if not isinstance(data, dict):
+        # Validate input using TelemetryData
+        try:
+            validated_data = TelemetryData.validate(data)
```

Ruff (0.14.10)

Local variable `validated_data` is assigned to but never used

Remove assignment to unused variable `validated_data`

(F841)